GraphQL.md

```diff
@@ -68,72 +91,141 @@ book {
+```
+author(ids: 1) {
+    id, 
+    books(op: REPLACE, data: [{id: 1, title: "New title"}]) {
+        id
+    }, 
+    publishers(op: REMOVE, id: 1) {
+        id
+    }
+}
+```
 # Semantics
-
-Below is a chart of expected behavior from GraphQL queries:
-
-| Operation | Id Parameter | Id in Body | Behavior |
-| --------- | ------------ | ---------- | -------- |
-| Add       | True         | True       |  Must match. Implies update. |
-|           | True         | False      | Update on persisted object with specified id |
-|           | False        | True       | Create new object (referenced by id in body within request) |
-|           | False        | False      | Create new object |
-| Replace   | True         | True       | Overwrite all specified values in body (including id) |
-|           | True         | False      | Overwrite all specified values (no id in body to overwrite) |
-|           | False        | True       | Remove from collection except if id exists, it won't be created |
-|           | False        | False      | Totally new collection values |
-| Fetch     | True         | True       | Boom. |
-|           | True         | False      | Find single id. |
-|           | False        | True       | Boom. |
-|           | False        | False      | Find all. |
-| Delete    | True         | True       | Must match. Remove single. |
-|           | True         | False      | Must match. Remove single. |
-|           | False        | True       | Remove matching id's |
-|           | False        | False      | Boom. |
-
-**NOTE:** If the id parameter is specified, it is always used as a **lookup** key for an already persisted object. Additionally, if the id parameter is specified outside of the data body, then the data must be a _single_ element list containing the proper object.
\ No newline at end of file
+While writing custom queries, you must take care of operations which do and do not allow the `data` and `ids` parameter fields as under.
+
+| Operation | Data | Ids |
+| --------- |------|----| 
+| Upsert    | ✓   | X   | 
+| Fetch     | X    | ✓    |
+| Replace   | ✓   | X     | 
+| Remove    | X   | ✓     | 
+| Delete    | X   | ✓      |
+**NOTE:** 
+* Creating objects with _UPSERT_ behave much like sql UPSERT, wherein, we first attempt to load the object, and if not present, we create it. 
+* If the id parameter is specified, it is always used as a **lookup** key for an already persisted object. Additionally, if the id parameter is specified outside of the data body, then the data must be a _single_ element list containing the proper object. 
+## graphQL Map Type 
+Elide supports [graphQL map query type](https://github.com/yahoo/elide/blob/add-data-fetcher/elide-core/src/main/java/com/yahoo/elide/graphql/GraphQLConversionUtils.java#L132). GraphQL inherently doesn't support map query types. We mimic
```

Comment:
I'm not sure what you mean by 'graphQL map query type'.  GraphQL doesn't have maps.